tests/user_examples/extra_src/shared/general_continuum/continuum_particles.cpp

```diff
@@ -42,5 +41,51 @@ namespace SPH
                          [&](size_t i) -> Vecd
                          { return n_[i]; });
     }
+    //=================================================================================================//
+    PlasticContinuumParticles::
+        PlasticContinuumParticles(SPHBody& sph_body, PlasticContinuum* plastic_continuum)
+        : ContinuumParticles(sph_body, plastic_continuum), plastic_continuum_(*plastic_continuum) {}
+    //=================================================================================================//
+    void PlasticContinuumParticles::initializeOtherVariables()
+    {
+        ContinuumParticles::initializeOtherVariables();
+
+        registerVariable(elastic_strain_tensor_3D_, "ElasticStrainTensor3D");
+        registerVariable(elastic_strain_rate_3D_, "ElasticStrainRate3D");
+        registerVariable(strain_tensor_3D_, "StrainTensor3D");
+        registerVariable(stress_tensor_3D_, "StressTensor3D");
+        registerVariable(strain_rate_3D_, "StrainRate3D");
+        registerVariable(stress_rate_3D_, "StressRate3D");
+        registerVariable(shear_stress_3D_, "ShearStress3D");
+        registerVariable(shear_strain_3D_, "ShearStrain3D");
+        registerVariable(shear_stress_rate_3D_, "ShearStressRate3D");
+        registerVariable(shear_strain_rate_3D_, "ShearStrainRate3D");
+        registerVariable(vertical_stress_, "VerticalStress");
+        registerVariable(acc_deviatoric_plastic_strain_, "AccDeviatoricPlasticStrain");
```

Comment:
Some dimension dependent variables are introduced. They can not be general?

Reply:
Actually, this is how we unify the 2D and 3D calculations for plastic materials. For plastic materials, we use the plane-strain assumption, but the stress should be in 3D no matter for 2D or 3D scenarios.